src/pools/abstract-pool.ts

```diff
@@ -202,14 +212,22 @@ export abstract class AbstractPool<
       throw new Error(
         'Cannot instantiate a pool with a negative number of workers'
       )
-    } else if (!this.dynamic && numberOfWorkers === 0) {
+    } else if (this.type === PoolType.FIXED && numberOfWorkers === 0) {
       throw new Error('Cannot instantiate a fixed pool with no worker')
     }
   }
 
+  private checkPoolOptions (opts?: PoolOptions<Worker>): void {
+    this.opts.enableEvents =
+      opts?.enableEvents !== undefined ? opts.enableEvents : true
```

Comment:
What if `enableEvents` would be `null`? Theoretically possible if ignoring TS while using JS